### PR TITLE
[cmake/getdns/jasper/mesa/sentry-native] Fix error C2065 and C2079

### PR DESCRIPTION
**Pull request by Cheney-W**
**Describe the pull request**
   In an internal version of VS, above ports install failed with following errors:
**cmake:**
`Source\kwsys\Terminal.c(94): error C2079: 'stream_stat' uses undefined struct 'stat'`
**getdns:**
`src\context.h(76): error C2079: 'prevstat' uses undefined struct 'stat'`
**jasper:**
`\src\libjasper\base\jas_stream.c(320): error C2065: 'O_RDWR': undeclared identifier`
`\src\libjasper\base\jas_stream.c(322): error C2065: 'O_RDONLY': undeclared identifier`
`\src\libjasper\base\jas_stream.c(324): error C2065: 'O_WRONLY': undeclared identifier`
`\src\libjasper\base\jas_stream.c(329): error C2065: 'O_APPEND': undeclared identifier`
`\src\libjasper\base\jas_stream.c(335): error C2065: 'O_CREAT': undeclared identifier`
`\src\libjasper\base\jas_stream.c(335): error C2065: 'O_TRUNC': undeclared identifier`
**mesa:**
`../src/esa-21.2.5-d0828d0837.clean/src/util/os_file.c(134): error C2065: 'O_RDONLY': undeclared identifier`
`../src/esa-21.2.5-d0828d0837.clean/src/util/os_file.c(143): error C2079: 'stat' uses undefined struct 'stat'`
`../src/esa-21.2.5-d0828d0837.clean/src/util/os_file.c(145): error C2224: left of '.st_size' must have struct/union type`
**sentry-native:**
`\src\path\sentry_path_windows.c(75): error C2065: 'LK_UNLCK': undeclared identifier`

This issue due to the macro `__STDC__` is defined in the new C language modes(/std:c11 or /std:c17) in this internal version of VS, but the CRT does not define those macros (and open()) because they are not standard C. 

According to the developers's comment, this issue could be workaround by adding the /D_CRT_DECLARE_NONSTDC_NAMES compile option.

### CODE REVIEW

**Comment by Neumann-A** [comment]
a) only required on windows (PR does not guard for that platform)
b) only required for newer (not published) VS versions?
c) current solution/fix does not respect custom toolchains which might not use `VCPKG_<LANG>_FLAGS `

**Comment by Cheney-W** [comment]
Yes, this issue only occurs in an internal VS version, user will not block by this issue currently, this workaround is just used to avoid blocking the internal CI test. 
I will report an issue to upstream and make a patch instead if I have a firm fix from compiler team.

**Review by BillyONeal — APPROVED** [review]

File: ports/getdns/portfile.cmake [hunk 1]
```diff
@@ -16,6 +16,11 @@ vcpkg_extract_source_archive(
     PATCHES disable-install-COPYING-in-config-step.patch
 )
 
+if(VCPKG_TARGET_IS_WINDOWS)
```
Comment (BillyONeal) [inline ports/getdns/portfile.cmake:19]: VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW?

/cc @dg0yt

Comment (dg0yt) [reply ports/getdns/portfile.cmake:19]: The macro probably wouldn't break anything but `/D` does, turning the parameter into a file path.
So `AND NOT VCPKG_TARGET_IS_MINGW` is mandatory here.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/getdns/portfile.cmake (hunk 1 above)
Comment (Cheney-W) [reply ports/getdns/portfile.cmake]: Thanks for your help!

**Review by LilyWangLL — APPROVED** [review]

**Comment by strega-nil** [comment]
Thanks!

**Pull request merged**